### ISSUE TITLE
-NoProfile should not add Modules folder under the profile in PSModulePath

### ISSUE THREAD

**Issue by ocalvo** [PowerShell/PowerShell#23957]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

1. Start cmd
2. Clear the environment variable PSModulePath
3. Start pwsh.exe -noprofile
4. Evaluate $env:PSModulePath

### Expected behavior

```console
PSModulePath should not include any path under the user profile.
```

### Actual behavior

```console
PSModulePath includes paths under the user profile.
```

### Error details

```console
N/A
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.2
PSEdition                      Core
GitCommitId                    7.4.2
OS                             Microsoft Windows 10.0.26236
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

N/A

**Comment by MartinGC94** [comment]
Why shouldn't it include those modules? It's not unreasonable to think someone would want to skip loading the `$profile` script(s) while still having those modules available. It would be better to ask for a new switch if you think it's useful to exclude user scoped modules.

**Comment by ocalvo** [comment]
Agree, will it be possible to introduce a new flag `-NoModulePath' ?

**Comment by ThomasNieto** [comment]
Can you please explain the use case for this feature?
